contracts/token/psp22/src/extensions/wrapper.rs

```diff
@@ -0,0 +1,84 @@
+/// Extension of [`PSP22`] which supports token wrapping
+use crate::traits::*;
+
+use brush::{
+    declare_storage_trait,
+    traits::{
+        AccountId,
+        Balance,
+        InkStorage,
+    },
+};
+use ink_prelude::vec::Vec;
+use ink_storage::traits::SpreadLayout;
+pub use psp22_derive::PSP22WrapperStorage;
+
+#[cfg(feature = "std")]
+use ink_storage::traits::StorageLayout;
+
+#[derive(Default, Debug, SpreadLayout)]
+#[cfg_attr(feature = "std", derive(StorageLayout))]
+pub struct PSP22WrapperData {
+    pub underlying: AccountId,
+}
+
+declare_storage_trait!(PSP22WrapperStorage, PSP22WrapperData);
+
+#[brush::wrapper]
+pub type PSP22WrapperWrapper = dyn PSP22WrapperExt + PSP22;
+
+#[brush::trait_definition]
+pub trait PSP22WrapperExt: PSP22WrapperStorage + PSP22 {
```

Comment:
You should use `PSP22Wrapper` here. Better then to rename all `TraitWrapper` into `TraitCaller` or `TraitCallBuilder`

Reply:
You also need to add a getter for `underlying`